main/src/cgeo/geocaching/filters/gui/DifficultyAndTerrainFilterViewHolder.java

```diff
@@ -0,0 +1,67 @@
+package cgeo.geocaching.filters.gui;
+
+import cgeo.geocaching.R;
+import cgeo.geocaching.filters.core.DifficultyAndTerrainGeocacheFilter;
+import cgeo.geocaching.filters.core.DifficultyGeocacheFilter;
+import cgeo.geocaching.filters.core.GeocacheFilterType;
+import cgeo.geocaching.filters.core.TerrainGeocacheFilter;
+import cgeo.geocaching.ui.TextParam;
+import cgeo.geocaching.ui.ViewUtils;
+
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.LinearLayout;
+
+import androidx.annotation.NonNull;
+
+public class DifficultyAndTerrainFilterViewHolder extends BaseFilterViewHolder<DifficultyAndTerrainGeocacheFilter> {
+
+    private IFilterViewHolder diffView = null;
+    private IFilterViewHolder terrainView = null;
+
+    @Override
+    public View createView() {
+        final LinearLayout ll = new LinearLayout(getActivity());
+        ll.setOrientation(LinearLayout.VERTICAL);
+
+        diffView = createItemRangeSelectorLayout(GeocacheFilterType.DIFFICULTY, TextParam.text("D"), ll);
+        terrainView = createItemRangeSelectorLayout(GeocacheFilterType.TERRAIN, TextParam.text("T"), ll);
+        ((ItemRangeSelectorViewHolder) terrainView).removeScaleLegend();
+
+        return ll;
+    }
+
+    @Override
+    public void setViewFromFilter(@NonNull final DifficultyAndTerrainGeocacheFilter filter) {
```

Comment:
DifficultyAndTerrainFilterViewHolder: 
warnings / remarks in AndroidStudio are shown for unchecked templates...
![image](https://user-images.githubusercontent.com/70113341/126655240-45557882-4999-4f37-bf51-4ae84db3ee39.png)

 I don't know how to get rid of it

Reply:
See my comment in line 20. I hope this helps. But if nothing helps, you can also add a `@SuppressWarnings("rawtypes")` to the method

Reply:
What does that warning mean? In what cases can that lead to an failure?

Reply:
Java allows for the usage of generic classes without specifying a class parameter (in this case it will assumed to be Object). This is called a "raw type". This is what you do with diffView and terrainView (you did not specify a class type using <>).

The method you call here however expects a parameter with a specified class type and uses this type. This can commonly lead to runtime exceptions because the raw class type may not be of the same class the method expects. But the compiler can no longer verify if this is the case or whether the code handles this correctly. This is why it issues this warning.

Im general, the usage of raw generic types should be avoided.